common/src/main/java/org/apache/drill/common/config/DrillConfig.java

```diff
@@ -44,31 +46,37 @@
 import com.google.common.collect.ImmutableList;
 import com.typesafe.config.Config;
 import com.typesafe.config.ConfigFactory;
+import com.typesafe.config.ConfigRenderOptions;
 
 public final class DrillConfig extends NestedConfig{
-//  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(DrillConfig.class);
+  private static final Logger logger = getLogger(DrillConfig.class);
+
   private final ObjectMapper mapper;
   private final ImmutableList<String> startupArguments;
 
   public static final boolean ON_OSX = System.getProperty("os.name").contains("OS X");
 
-  @SuppressWarnings("restriction")  private static final long MAX_DIRECT_MEMORY = sun.misc.VM.maxDirectMemory();
+  @SuppressWarnings("restriction")
+  private static final long MAX_DIRECT_MEMORY = sun.misc.VM.maxDirectMemory();
 
   @SuppressWarnings("unchecked")
   private volatile List<Queue<Object>> sinkQueues = new CopyOnWriteArrayList<Queue<Object>>(new Queue[1]);
 
   @VisibleForTesting
-  public DrillConfig(Config config, boolean enableServer) {
+  public DrillConfig(Config config, boolean enableServerConfigs) {
     super(config);
-    logger.debug("Setting up config object.");
+    logger.debug("Setting up DrillConfig object.");
+    if (logger.isTraceEnabled()) {
```

Comment:
Again I'm not particularly opinionated on this one. If we are trying to establish best practices, I would argue for leaving out the guard unless you are confident enough about the need for it that you would put in a comment describing the overhead you are avoiding. Otherwise we will have inconsistent patterns, and someone might find them and assume we always want the guards.